docs/design/policy-result-identity.md

```diff
@@ -0,0 +1,120 @@
+# Policy result identity
+
+- [Policy result identity](#policy-result-identity)
+  - [Background](#background)
+  - [Suggested change](#suggested-change)
+  - [The `result_tag` attribute](#the-result_tag-attribute)
+  - [Potential risk](#potential-risk)
+
+We have identified a need for some policies to produce multiple, distinct
+results for a single resource. This document is a proposal for an enhancement to
+an existing mechanism in Policy Engine that achieves this.
+
+## Background
+
+Policy Engine currently combines `deny` and `resources` results together based
+on a ["correlation ID"](https://github.com/snyk/policy-engine/blob/5e55e9bc644c35aebaf82defd84abf7cb7f97f92/docs/policy_spec.md#correlation-ids).
+The default correlation ID is based on the primary resource's namespace, type,
+and ID. This produces a single result per resource, per policy.
+
+Policy authors are able to supply their own correlation ID in order to change
+the way that results are grouped. This mechanism is very flexible and enables
+policies to produce multiple results for a single resource or a single result
+for multiple resources.
+
+The drawback to this mechanism is that it's not communicated to or understood by
+our downstream consumers, which are built for a handful of known use-cases:
+
+- A single primary resource
+- A missing resource
+- A failing attribute within a resource
+
+## Suggested change
+
+Because a single result for multiple resources isn't a current use-case, we
+propose to:
+
+- Remove `correlation` from our Rego API
+  - Policy Engine will still use this mechanism internally, but it shouldn't be
+    exposed to policy authors.
+- Add a `result_tag` attribute which can be used to uniquely identify a result
+  when multiple results are produced for a single resource.
+- Include the `result_tag` attribute in our output format
+
+This accomplishes three goals:
+
+- Removes the ability to produce a single result for multiple resources, because
```

Comment:
```suggestion
- Removes the ability to produce a single result for multiple primary resources, because
```

Not sure if my suggestion makes it more confusing, but I think the distinction is important?  Maybe something like:

> Removes the ability to group multiple issues, each with distinct primary resources, under one issue.